semanticdb/semanticdb3/javacp.md

```diff
@@ -0,0 +1,551 @@
+### Java
+
+In this section, we exhaustively map Java language features onto SemanticDB.
+As a reference, we use the Java Language Specification [\[85\]][85] (referred
+to as "JLS" in the text below).
+
+<a name="java-symbol"></a>
+#### Symbol
+
+In this section, we describe the Java symbol format.
+
+<table>
+  <tr>
+    <td><b>Symbols</b></td>
+    <td><b>Format</b></td>
+  </tr>
+  <tr>
+    <td>
+      Global symbols
+      <a href="#symbol">↑</a>
+    </td>
+    <td>
+      <ul>
+        <li>
+          For <a href="#java-root-package">root package</a>, its descriptor.
+        </li>
+        <li>
+          For unnamed packages <a href="https://docs.oracle.com/javase/specs/jls/se8/html/jls-7.html#jls-7.4.2">[21]</a>,
+          its descriptor.
+        </li>
+        <li>
+          For top-level package, its descriptor.
+        </li>
+        <li>
+          For other definition, concatenation of owner symbol and
+          definition descriptor.
+        </li>
+      </ul>
+    </td>
+  </tr>
+  <tr>
+    <td>
+      Local symbols
+      <a href="#symbol">↑</a>
+    </td>
+    <td>
+      Concatenation of <code>local</code> and a decimal number.
+    </td>
+  </tr>
+</table>
+
+**Owner** is:
+  * For root package, `None`.
+  * For unnamed package, root package.
+  * For top-level named package, root package.
+  * For other named package, enclosing package that holds the named package as
+    a member.
+  * For other top-level declaration, innermost enclosing package that holds
+    the top-level declaration as a member.
+  * For other other global definition, innermost enclosing declaration that
+    holds the global declaration as a member.
+  * For other declarations, `None`.
+
+**Descriptor** is:
+  * For `FIELD` or `PACKAGE`, concatenation of its simple name and a dot (`.`).
+  * For `METHOD` or `CONSTRUCTOR`, concatenation of its identifier, a disambiguator, and a dot (`.`).
+  * For `CLASS` or `INTERFACE`, concatenation of its identifier and a pound sign (`#`).
+  * For `PARAMETER`, concatenation of a left parenthesis (`(`), its identifier and a right parenthesis (`)`).
+  * For `TYPE_PARAMETER`, concatenation of a left bracket (`[`), its identifier and a right bracket (`]`).
+  * See [SymbolInformation](#java-symbolinformation) for details on
+    which Scala definitions are modelled by which symbols.
+
+**Disambiguator** is:
+  * Concatenation of a left parenthesis (`(`), a type descriptor
+    and a right parenthesis (`)`).
+    In the case when multiple definitions have the same kind, name and
+    type descriptor, the type descriptor is appended with `+N`,
+    with `+1` going to the method that is defined first in the source code,
+    `+2` going to the method that is defined second, etc.
+
+**Name** is:
+  * For root package, `_root_`.
+  * For unnamed package, `_empty_`.
+  * For named package, the latest identifier of its package declaration.
```

Comment:
How about merging the bullet about packages and the bullet about other definitions?